### PR TITLE
Aaust fmwpc reco2

### PR DESCRIPTION
**Pull request by aaust**
* write FMWPC hits to REST file
* remove quality cuts
* get threshold and timing cut from CCDB or commandline

### CODE REVIEW

File: src/libraries/FMWPC/DFMWPCHit_factory.cc [hunk 1]
```diff
@@ -66,9 +66,35 @@ jerror_t DFMWPCHit_factory::brun(jana::JEventLoop *eventLoop, int32_t runnumber)
...
+
+  gPARMS->SetDefaultParameter("FMWPC:FMWPC_HIT_THRESHOLD", hit_threshold,
+                              "Remove FMWPC Hits with peak amplitudes smaller than FMWPC_HIT_THRESHOLD");
+
+  vector<double> fmwpc_timing_cuts;
+
+  if (eventLoop->GetCalib("/FMWPC/timing_cut", fmwpc_timing_cuts)){
+    t_raw_min = -60.;
+    t_raw_max = 900.;
+    jout << "Error loading /FMWPC/timing_cut ! set defaul values -60. and 900." << endl;
```
Comment (sdobbs) [inline src/libraries/FMWPC/DFMWPCHit_factory.cc:88]: typo here

**Comment by gluex** [comment]
Test status for this pull request: FAILURE
 
 Failures: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/failures.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-failures.txt)
 Summary: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log)
 Build report: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log)
 Build report: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2

**Comment by aaust** [comment]
The early failures of the test were caused by a faulty version set.

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/make_aaust_fmwpc_reco2.log)
 Build report: [/work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^aaust_fmwpc_reco2/report_aaust_fmwpc_reco2.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^aaust_fmwpc_reco2

**Comment by aaust** [comment]
A quick test with 100k events showed only a 5% increase in the size.

**Review by sdobbs — APPROVED** [review]
looks good to have the information there.  increase in REST size sounds reasonable

**Pull request merged**